src/ImageProcessorCore/Numerics/Rational.cs

```diff
@@ -0,0 +1,1143 @@
+﻿// --------------------------------------------------------------------------------------------------------------------
+// <copyright file="Rational.cs" company="James South">
+//   Copyright (c) James South.
+//   Licensed under the Apache License, Version 2.0.
+// </copyright>
+// <summary>
+//   Represents a rational number. Any number that can be expressed as the quotient or fraction p/q of two
+//   numbers, p and q, with the denominator q not equal to zero.
+//   <remarks>
+//   Adapted from <see href="https://github.com/mckamey/exif-utils.net" /> by Stephen McKamey.
+//   </remarks>
+// </summary>
+// --------------------------------------------------------------------------------------------------------------------
+
+namespace ImageProcessorCore
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics.CodeAnalysis;
+    using System.Globalization;
+    using System.Reflection;
+
+    /// <summary>
+    /// Represents a rational number. Any number that can be expressed as the quotient or fraction p/q of two 
+    /// numbers, p and q, with the denominator q not equal to zero.
+    /// <remarks>
+    /// Adapted from <see href="https://github.com/mckamey/exif-utils.net"/> by Stephen McKamey.
+    /// </remarks>
+    /// </summary>
+    /// <typeparam name="T">
+    /// The type to assign to the numerator and denominator components.
+    /// </typeparam>
+    [SuppressMessage("StyleCop.CSharp.OrderingRules", "SA1202:ElementsMustBeOrderedByAccess", Justification = "Reviewed. Suppression is OK here. Better readability.")]
+    public struct Rational<T> :
+        IConvertible,
```

Comment:
It looks like Microsoft has removed this Interface "IConvertible" from the standard library. Its also missing the TypeCode enumeration used by this class.  Any Idea's why or what package we can include that has it.